plugins/browser-plugin-ga-cookies/src/index.ts

```diff
@@ -1,54 +1,85 @@
- * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
- * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
- */
-
-import { SelfDescribingJson } from '@snowplow/tracker-core';
+import { LOG, SelfDescribingJson } from '@snowplow/tracker-core';
 import { BrowserPlugin, cookie } from '@snowplow/browser-tracker-core';
-import { Cookies } from './contexts';
+import { GA4Cookies, UACookies } from './types';
+import { GOOGLE_ANALYTICS_4_COOKIES_SCHEMA, UNIVERSAL_ANALYTICS_COOKIES_SCHEMA } from './schemata';
+
+interface GACookiesPluginOptions {
+  ua?: boolean;
+  ga4: boolean;
+  ga4MeasurementId: string | string[];
+  cookiePrefix?: string;
+}
+
+const defaultPluginOptions: GACookiesPluginOptions = {
+  ua: true,
+  ga4: false,
+  ga4MeasurementId: '',
+  cookiePrefix: '',
+};
 
 /**
  * Captures the GA cookies on a page and sends as context on each event
+ * @param pluginOptions.ua - Send Universal Analytics specific cookie values.
+ * @param pluginOptions.ga4 - Send Google Analytics 4 specific cookie values.
+ * @param pluginOptions.ga4MeasurementId - Measurement id/ids to search the Google Analytics 4 session cookie. Can be a single measurement id as a string or an array of measurement id strings. The cookie has the form of _ga_<container-id> where <container-id> is the data stream container id.
+ * @param pluginOptions.cookiePrefix - Cookie prefix set on the Google Analytics 4 cookies.
  */
-export function GaCookiesPlugin(): BrowserPlugin {
+export function GaCookiesPlugin(pluginOptions: GACookiesPluginOptions = defaultPluginOptions): BrowserPlugin {
   return {
     contexts: () => {
-      const gaCookieData: SelfDescribingJson<Cookies> = {
-        schema: 'iglu:com.google.analytics/cookies/jsonschema/1-0-0',
-        data: {},
-      };
-      ['__utma', '__utmb', '__utmc', '__utmv', '__utmz', '_ga'].forEach(function (cookieType) {
-        var value = cookie(cookieType);
-        if (value) {
-          gaCookieData.data[cookieType] = value;
+      const contexts: SelfDescribingJson<Record<string, unknown>>[] = [];
+      const { ga4, ga4MeasurementId, ua, cookiePrefix } = { ...defaultPluginOptions, ...pluginOptions };
+      const GA_USER_COOKIE = '_ga';
+      const GA4_MEASUREMENT_ID_PREFIX = 'G-';
+      const GA4_COOKIE_PREFIX = '_ga_';
+
+      if (ua) {
+        const uaCookiesContext: SelfDescribingJson<UACookies> = {
+          schema: UNIVERSAL_ANALYTICS_COOKIES_SCHEMA,
+          data: {},
+        };
+        ['__utma', '__utmb', '__utmc', '__utmv', '__utmz', GA_USER_COOKIE].forEach(function (cookieType) {
+          var value = cookie(cookieType);
+          if (value) {
+            uaCookiesContext.data[cookieType] = value;
+          }
+        });
+        contexts.push(uaCookiesContext);
+      }
+
+      if (ga4) {
+        if (!ga4MeasurementId) {
+          LOG.warn('"ga4MeasurementId" option not provided in browser-plugin-ga-cookies.');
+          return contexts;
```

Comment:
So the API is now that if you want the old behaviour (client ID cookie but don't care about session IDs), you need to specify `ga4MeasurementId: []` to pass this check. Should we be more explicit that that's an option?

Also the default now for `{ua: false, ga4: true}` will do nothing, is that right?

Reply:
Updated to send the user cookie (_ga) when `{ ga4: true }` and not log the warning.
Should make sure the documentation page specifies that if you do not pass measurement ids, you would not be able to get session cookies.